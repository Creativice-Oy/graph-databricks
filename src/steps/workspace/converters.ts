import {
  createIntegrationEntity,
  Entity,
} from '@jupiterone/integration-sdk-core';

import { Entities } from '../constants';

export function createWorkspaceEntity(subdomain: string): Entity {
  return createIntegrationEntity({
    entityData: {
      source: {
        id: subdomain,
      },
      assign: {
        _key: `databricks_workspace:${subdomain}`,
        _type: Entities.WORKSPACE._type,
        _class: Entities.WORKSPACE._class,
        name: subdomain,
      },
    },
  });
}
