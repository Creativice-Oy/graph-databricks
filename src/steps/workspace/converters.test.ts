import { getMockWorkspace } from '../../../test/mocks';
import { createWorkspaceEntity } from './converters';

describe('#createWorkspaceEntity', () => {
  test('should convert to entity', () => {
    expect(createWorkspaceEntity(getMockWorkspace())).toMatchSnapshot();
  });
});
