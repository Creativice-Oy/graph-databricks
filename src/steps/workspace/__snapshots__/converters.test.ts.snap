// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createWorkspaceEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Account",
  ],
  "_key": "databricks_workspace:3840207185082406.6",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "id": "3840207185082406.6",
      },
    },
  ],
  "_type": "databricks_workspace",
  "createdOn": undefined,
  "displayName": "3840207185082406.6",
  "id": "3840207185082406.6",
  "name": "3840207185082406.6",
}
`;
